common/src/main/java/org/conscrypt/AbstractSessionContext.java

```diff
@@ -120,51 +74,63 @@ public final int getSessionTimeout() {
     }
 
     /**
+     * Object used for locking access to the sessions.
+     */
+    abstract Object sessionsLock();
```

Comment:
Yeah it's admittedly ugly.  Alternatively, we could synchronize on the class.  WDYT?